nym-vpn-core/crates/nym-vpn-lib-types/src/tunnel_state.rs

```diff
@@ -198,6 +205,117 @@ pub enum ErrorStateReason {
     Internal(String),
 }
 
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub enum ClientErrorReason {
+    Firewall,
+    Routing,
+    SameEntryAndExitGateway,
+    InvalidEntryGatewayCountry,
+    InvalidExitGatewayCountry,
+    MaxDevicesReached,
+    BandwidthExceeded,
+    SubscriptionExpired,
+    Dns(String),
+    Api(String),
+    Internal(String),
+}
+
+impl From<ErrorStateReason> for ClientErrorReason {
+    fn from(value: ErrorStateReason) -> Self {
+        match value {
+            ErrorStateReason::SameEntryAndExitGateway => Self::SameEntryAndExitGateway,
+            ErrorStateReason::InvalidEntryGatewayCountry => Self::InvalidEntryGatewayCountry,
+            ErrorStateReason::InvalidExitGatewayCountry => Self::InvalidExitGatewayCountry,
+            ErrorStateReason::BadBandwidthIncrease => Self::Api(value.to_string()),
+            ErrorStateReason::SyncAccount(err) => Self::Api(err.to_string()),
```

Comment:
I'm not sure all SyncAccount errors can be translated as "Api errors", for example what about `SyncAccountErr::NoAccountStored` and `SyncAccountErr::Internal`?
Also with this simplified mapping, we're losing the error origin, ie. from "SyncAccount".

Reply:
let's compare whit how I implemented it in tauri side -> https://github.com/nymtech/nym-vpn-client/blob/73086d6510ebd5a3cf2ecc1b927e9cb8cb7c7f01/nym-vpn-app/src-tauri/src/grpc/tunnel_error.rs#L106